packages/react-sdk/docusaurus/docs/React/02-guides/03-call-and-participant-state.mdx

```diff
@@ -3,89 +3,193 @@ title: Call & Participant State
 description: How the state is exposed
 ---
 
-:::warning
-
-TODO: add better code snippets
-:::
-
 Our SDK uses a design pattern that we refer to as a reactive state store.
 
-The `StreamVideoClient` instance manages its own state store (this contains information about the connected user and the list of incoming, outgoing and active calls) and each `Call` instance has its own state store (this contains information about the call itself, for example the list of participants) as well.
+The [`StreamVideoClient`](../../call-engine/StreamVideoClient/) instance manages its own state store (this contains information about the connected user and the list of incoming, outgoing and active calls). Additionally each [`Call`](../../call-engine/Call/) instance has its own state store (this contains information about the call itself, for example, the list of participants).
 
-The state stores are updated after each API call/WebSocket event. The reactive state store makes it possible to observe the state from anywhere in your application. You can access the state store using hooks. This guide will help you understand how state management work which is useful if you want to attach custom logic upon state changes or if you want to create custom components.
+The state stores are updated after each API call/WebSocket event. The reactive state store makes it possible to observe the state from anywhere in your application. You can access the state store using hooks. This guide will help you understand how state management works which is helpful if you want to attach custom logic upon state changes or if you want to create custom components.
 
 ## Create client
 
-Before we can explore the state management mechanism we need to have a `StreamVideoClient` instance, to create it we can use the [`useCreateStreamVideoClient`](../../call-engine/hooks-and-contexts#usecreatestreamvideoclient) hook.
+Before we can explore the state management mechanism, we need to have a `StreamVideoClient` instance, to create it we can use the [`useCreateStreamVideoClient`](../../call-engine/hooks-and-contexts#usecreatestreamvideoclient) hook.
 
-```typescript
-const client = useCreateStreamVideoClient({
-    <your API key>,
-    <user token>,
-    <user>,
-});
+```tsx
+import {
+  useCreateStreamVideoClient,
+  StreamVideo,
+  useConnectedUser,
+} from '@stream-io/video-react-sdk';
+
+export default function App() {
+  const client = useCreateStreamVideoClient({
+    apiKey: '<Your API key>',
+    tokenOrProvider: '<token provider>',
+    user: {
+      id: '<user id>',
+      name: '<name>',
+    },
+  });
+  return ();
```

Comment:
```suggestion
  return <StreamVideo client={client}><MyApp /></StreamVideoClient>;
```

Reply:
The `StreamVideo` context is added in the next step